### PR TITLE
chore(overflow): Instrument on capacity overflow in plugin-server

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

Ingestion capacity overflow detection is deployed on the Plugin Server but we have little visilibity on how this is working.

## Changes

Adds logging and a counter to track overflows. Important that both fire off only once a minute to avoid overflowing metric servers. This is the same behavior we have in the capture endpoint.

### CODE REVIEW

**Comment by tomasfarias** [comment]
@xvello You may also be interested as you have been working with metrics and in particular added the overflow metric on the capture side.

File: plugin-server/src/main/ingestion-queues/analytics-events-ingestion-consumer.ts [hunk 1]
```diff
@@ -91,6 +98,11 @@ export async function eachBatchIngestionWithOverflow(
                 // Set message key to be null so we know to send it to overflow topic.
                 // We don't want to do it here to preserve the kafka offset handling
                 message.key = null
+
+                if (LoggingLimiter.consume(seenKey, 1) === true) {
+                    ingestionPartitionKeyOverflowed.labels(seenKey).inc()
```
Comment (hazzadous) [inline plugin-server/src/main/ingestion-queues/analytics-events-ingestion-consumer.ts:103]: Can we put this outside of the if? We aren't concerned about limiting the amount we record in metrics I don't think.

**Review by hazzadous — APPROVED** [review]
Looks good.

Note that we can see which teams have hit this with [this query](https://metabase.posthog.net/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJzZWxlY3QgdGVhbV9pZCwgSlNPTkV4dHJhY3RTdHJpbmcoZGV0YWlscywgJ292ZXJmbG93RGlzdGluY3RJZCcpIGRpc3RpbmN0X2lkLCBjb3VudCgqKSBjXG5mcm9tIGluZ2VzdGlvbl93YXJuaW5nc1xud2hlcmUgdHlwZSA9ICdpbmdlc3Rpb25fY2FwYWNpdHlfb3ZlcmZsb3cnXG5hbmQgdGltZXN0YW1wID49ICcyMDIzLTAyLTIzJ1xuZ3JvdXAgYnkgdGVhbV9pZCwgZGlzdGluY3RfaWRcbm9yZGVyIGJ5IGNcblxuXG5zZWxlY3QgKiBcbmZyb20gaW5nZXN0aW9uX3dhcm5pbmdzXG53aGVyZSB0eXBlID0gJ2luZ2VzdGlvbl9jYXBhY2l0eV9vdmVyZmxvdydcbmFuZCB0aW1lc3RhbXAgPj0gJzIwMjMtMDItMjMnXG5saW1pdCAxMCIsInRlbXBsYXRlLXRhZ3MiOnt9fSwiZGF0YWJhc2UiOjJ9LCJkaXNwbGF5IjoidGFibGUiLCJwYXJhbWV0ZXJzIjpbXSwidmlzdWFsaXphdGlvbl9zZXR0aW5ncyI6e319)

We could probably do with making the bucket a little larger or leaker. Not sure which one.

File: plugin-server/src/main/ingestion-queues/analytics-events-ingestion-consumer.ts (hunk 1 above)
Comment (tomasfarias) [reply plugin-server/src/main/ingestion-queues/analytics-events-ingestion-consumer.ts:103]: Of course 👍

**Pull request merged**